File: lesson01/code/criarFrase.py
def criarFrase(palavras, separador=' '):
    frase = palavras[0]
    for i in range(1, len(palavras)):
        frase = frase + separador + palavras[i]

    return frase

palavras = ['Criando', 'uma', 'frase']
print(criarFrase(palavras))
print(criarFrase(palavras, ' '))
print(criarFrase(palavras, '-'))

File: lesson01/code/CriarFrase.java
public class CriarFrase
{
    public static String criarFrase(String[] palavras)
    {
        return CriarFrase.criarFrase(palavras, ' ');
    }

    public static String criarFrase(String[] palavras, char separador)
    {
        String frase = palavras[0];
        for(int i = 1; i < palavras.length; i++)
        {
            frase = frase + separador + palavras[i];
        }

        return frase;
    }

    public static void main(String[] args)
    {
        String[] palavras = {"Criando", "uma", "frase"};
        System.out.println(CriarFrase.criarFrase(palavras));
        System.out.println(CriarFrase.criarFrase(palavras, ' '));
        System.out.println(CriarFrase.criarFrase(palavras, '-'));
    }
}